### DISCUSSION TITLE
Managing multiple roles

### DISCUSSION THREAD

**Discussion by gmwill934** [prisma/prisma#20872] [Q&A]
### Question

Hello everyone,

I'm wondering if there's a way to effectively handle multiple roles (Postgresql users) using Prisma. In my case, I have row-level security (`RLS`) enabled for all of my tables. However, I'm facing the challenge of needing to bypass these policies temporarily when a user is in the process of logging in or registering. Any insights or suggestions on how to approach this would be greatly appreciated.

**Comment by gmwill934** [comment]
Is there a way to override this? 👀 I'm pretty sure a lot of people could benefit from a feature like this.

**Reply by ludralph** [reply]
Hi @gmwill934 👋 

You can use Prisma Client Extensions to handle row-level security (RLS) in Postgres. Here is an [example](https://www.prisma.io/blog/client-extensions-preview-8t3w27xkrxxn#example-row-level-security) using Client Extensions for RLS.  This can be particularly useful in a multi-tenant application where you need to isolate data between tenants.

Here's an [example](https://github.com/prisma/prisma-client-extensions/tree/main/row-level-security#4-create-row-security-policies-to-bypass-rls-optional) of how you can use Prisma Client Extensions to bypass RLS:
```ts
function bypassRLS() {
  return Prisma.defineExtension((prisma) =>
    prisma.$extends({
      query: {
        $allModels: {
          async $allOperations({ args, query }) {
            const [, result] = await prisma.$transaction([
              prisma.$executeRaw`SELECT set_config('app.bypass_rls', 'on', TRUE)`,
              query(args),
            ]);
            return result;
          },
        },
      },
    })
  );
}
```

In this example, the `bypassRLS` function is used to bypass RLS. This function can be called when a user is in the process of logging in or registering, allowing you to bypass your RLS policies temporarily.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
